windows/flutter_nsd_plugin.cpp

```diff
@@ -0,0 +1,335 @@
+  }
+
+  void MdnsRequest::runner(std::string serviceName) {
+    while (keepRunning) {
+      DWORD result = mdns_query(this, (serviceName + "local.").c_str());
+      for (auto it = packets.begin(); it != packets.end(); it++) {
+        send(it->second);
+      }
+      if (result == -1) return;
+    }
+    delete this;
+  }
+
+
+  void MdnsRequest::ThreadFunc(MdnsRequest *p, std::string serviceName)
+  {
+    p->runner(serviceName);
+  }
+
+
+  void MdnsRequest::start(std::string serviceName)
+  {
+    keepRunning = TRUE;
+    std::thread myThread(ThreadFunc, this, serviceName);
+    myThread.detach();
+  }
+
+  void MdnsRequest::stop()
+  {
+    keepRunning = false;
+  }
+
+
+class FlutterNsdPlugin : public flutter::Plugin {
+ public:
+  static void RegisterWithRegistrar(flutter::PluginRegistrarWindows *registrar);
+
+  FlutterNsdPlugin(std::shared_ptr<flutter::MethodChannel<flutter::EncodableValue>>);
+
+  virtual ~FlutterNsdPlugin();
+
+ private:
+  // Called when a method is called on this plugin's channel from Dart.
+  void HandleMethodCall(
+      const flutter::MethodCall<flutter::EncodableValue> &method_call,
+      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result);
+  std::shared_ptr<flutter::MethodChannel<flutter::EncodableValue>> channel;
+
+  MdnsRequest* currentRequest;
+
+};
+
+// static
+void FlutterNsdPlugin::RegisterWithRegistrar(
+    flutter::PluginRegistrarWindows *registrar) {
+  auto channel =
+      std::make_shared<flutter::MethodChannel<flutter::EncodableValue>>(
+          registrar->messenger(), "com.nimroddayan/flutter_nsd",
+          &flutter::StandardMethodCodec::GetInstance());
+
+  auto plugin = std::make_unique<FlutterNsdPlugin>(channel);
+
+  channel->SetMethodCallHandler(
+      [plugin_pointer = plugin.get()](const auto &call, auto result) {
+        plugin_pointer->HandleMethodCall(call, std::move(result));
+      });
+
+  registrar->AddPlugin(std::move(plugin));
+}
+
+
+
+FlutterNsdPlugin::FlutterNsdPlugin(std::shared_ptr<flutter::MethodChannel<flutter::EncodableValue>> _channel)
+
+{
+  channel = _channel;
+  currentRequest = NULL;
+
+  WORD versionWanted = MAKEWORD(1, 1);
+  WSADATA wsaData;
+  if (WSAStartup(versionWanted, &wsaData)) {
+
+    printf("Failed to initialize WinSock\n");
+  }
+}
+
+FlutterNsdPlugin::~FlutterNsdPlugin() {
+  WSACleanup();
+}
+
+
+
+void FlutterNsdPlugin::HandleMethodCall(
+  const flutter::MethodCall<flutter::EncodableValue>& method_call,
+  std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
+  if (method_call.method_name().compare("getPlatformVersion") == 0) {
+    std::ostringstream version_stream;
+    version_stream << "Windows ";
+    if (IsWindows10OrGreater()) {
+      version_stream << "10+";
+    }
+    else if (IsWindows8OrGreater()) {
+      version_stream << "8";
+    }
+    else if (IsWindows7OrGreater()) {
+      version_stream << "7";
+    }
+    result->Success(flutter::EncodableValue(version_stream.str()));
+  }
```

Comment:
This should be removed as the plugin doesn't support such method. Only `startDiscovery` and `stopDiscovery`.

Reply:
If you want I can add support for these calls to the other platforms